rfc/plugins.md

```diff
@@ -0,0 +1,235 @@
+# Plugins
+
+A plugin is an ember-cli addon that hooks into the ember-cli-deploy
+pipeline in order to add functionality to a deploy. Example addon
+functionality would be uploading assets to S3, writing the index.html to
+Redis, or notifying a Slack channel a deploy has completed.
+
+Plugins should focus on doing a specific task. Ember developers will
+compose and combine multiple plugins to fine tune their deployment
+process.
+
+### Identifying plugins
+
+Plugins are published to npm as ember-cli addons. They will also have
+the keyword ``ember-cli-deploy-plugin``.
+
+Any plugin that is installed will be loaded by default, similar to
+ember-cli addons.
+
+### Base plugin object
+
+Plugins will extend a base object provided by ember-cli-deploy. The API
+of this object will be small to start, but it will give us an easy place
+to give these plugins additional interfaces if needed.
+
+```javascript
+S3Plugin = DeployPlugin.extend({
+  ...
+});
+```
+
+### Hooks
+
+Plugins hook into the deployment pipeline with function names that match
+the steps in the pipeline process. These hooks will scoped under a
+special hooks object inside the plugin so that any private or plugin
+functions do not clash with any hook names that we add in
```

Comment:
The `hooks` object is going to make method context weird. How about of instead the properties names for the functions are prefixed with `deploy:`. e.g.

```javascript
var S3Plugin = DeployPlugin.extend({
  "deploy:updateAssets": function(...){
    ...
  },
  "deploy:activate": function(...){
    ...
  },
});
```